### DISCUSSION TITLE
A procedure imported by 'CefSharp.Core.Runtime.dll' could not be loaded.

### DISCUSSION THREAD

**Discussion by GitNewbieHub** [cefsharp/CefSharp#4593] [General]
I'm upgrading CefSharp from 49.0.1.0 to 116.0.230.0 for one of my Winform project(dll). Everything works fine locally. I copied my updated dll file and all the CefSharp related files to a test server. I'm getting  "A procedure imported by 'CefSharp.Core.Runtime.dll' could not be loaded." I noticed 49.0.1.0 doesn't have CefSharp.Core.Runtime.dll yet. so CefSharp.Core.Runtime.dll is not registered. How can I register  CefSharp.Core.Runtime.dll . I'm getting The module "CefSharp.Core.Runtime.dll  failed to load. Make sure the binary is stored at the specified path or debug it to check for problem with the binary of depend .DLL files." when try to run regsvr32

Machine is windows server 2012 and all the required files are present.
https://github.com/cefsharp/CefSharp/wiki/Output-files-description-table-%28Redistribution%29

Updated:

So since Windows Server 2012 is no longer supported in the later versions.

I just upgraded/downgraded CefSharp to 109.1.11 and ran into a different issue. I'm getting application has stopped error

Faulting application name: MyApp.Client.exe, version: 22.4.0.0, time stamp: 0x6510f3ee
Faulting module name: libcef.dll, version: 109.1.11.0, time stamp: 0x63b8965c
Exception code: 0x4000001f
Fault offset: 0x01587b28
Faulting process id: 0x254
Faulting application start time: 0x01d9f09d35ab820c
Faulting application path: C:\Program Files (x86)\MyApp\MyApp.Client.exe
Faulting module path: C:\Program Files (x86)\MyApp\libcef.dll
Report Id: 9b061e52-5c90-11ee-8198-000d3a4632ed
Faulting package full name:
Faulting package-relative application ID:

Error occurred on following line code

Cef.Initialize(settings);

Here is the log from debug.log

[0926/195836.784:ERROR:dxva_video_decode_accelerator_win.cc(1460)] DXVAVDA fatal error: could not LoadLibrary: msmpeg2vdec.dll: The specified module could not be found. (0x7E)
[0926/195836.784:ERROR:dxva_video_decode_accelerator_win.cc(1468)] DXVAVDA fatal error: could not LoadLibrary: msvproc.dll: The specified module could not be found. (0x7E)

I copied the two files from my Windows 10 machine to the server's system32 folder but it didn't resolve the error. Also tried different version(109.0.9.0), same issue. I couldn't find much info on google also.

**Comment by n303ef** [comment]
Windows Server versions up to 2012 are not supported since CefSharp 110 as well as Google Chrome. You need Windows Server 2016 or higher.

**Reply by GitNewbieHub** [reply]
Where did you see this? I actually looked it up and but couldn't find anything. Only found window 7 is not supported.

**Reply by amaitland** [reply]
> Machine is windows server 2012 and all the required files are present.

No longer supported.

> Where did you see this? I actually looked it up and but couldn't find anything. Only found window 7 is not supported.

Server 2012 is the same code base as Windows 7

Read the Chrome announcement link in #4336

**Reply by GitNewbieHub** [reply]
Thanks!, I ran into a different issue now after upgraded to the last version that supports Windows 7/Windows Server 2012.

**Reply by amaitland** [reply]
Are you running a cut down version of windows server? e.g. essentials? Try installing the media pack.

> I copied the two files from my Windows 10 machine to the server's system32 folder but it didn't resolve the error

They're fundamentally different operating systems, you'll need to **install** the correct version.

You might be able to disable the required chromium component, https://stackoverflow.com/a/72391063/4583726

Your using an old unsupported version, on an unsupported OS. Upgrading both is the approach I'd recommend.

**Reply by GitNewbieHub** [reply]
@amaitland  Thank you very much. I got the servers upgraded to 2016. It works fine on one of server but on another server it doesn't load(WinForm). It has the following errors logged in the debug.log

[1017/212645.026:WARNING:policy_logger.cc(148)] :components\enterprise\browser\controller\chrome_browser_cloud_management_controller.cc(88) Could not create policy manager as CBCM is not enabled.

[1017/212645.376:ERROR:ssl_client_socket_impl.cc(968)] handshake failed; returned -1, SSL error code 1, net_error -207


I'm assuming it's the ERROR caused the issue? but why only on one machine?

**Reply by GitNewbieHub** [reply]
Figured it out. The machine has invalid certificate. Add settings.IgnoreCertificateErrors = true resolved the issue.
